tslearn/svm.py

```diff
@@ -197,25 +199,28 @@ def support_vectors_time_series_(self, X):
         return sv
 
     def fit(self, X, y, sample_weight=None):
-        X, y = check_X_y(X, y, allow_nd=True)
+        if self.kernel in self.variable_length_kernels:
+            X, y = check_X_y(X, y, allow_nd=True, force_all_finite=False)
+        else:
+            X, y = check_X_y(X, y, allow_nd=True)
         X = check_dims(X, X_fit=None)
+        X = to_time_series_dataset(X)
 
         self._X_fit = X
         self.classes_ = numpy.unique(y)
 
-        _, sz, d = X.shape
-        sklearn_X = _prepare_ts_datasets_sklearn(X)
-
-        gamma = self.gamma
-        kernel = self.kernel
-        if gamma == "auto":
-            gamma = gamma_soft_dtw(to_time_series_dataset(X))
-        if kernel == "gak":
-            kernel = GAKKernel(sz, d, gamma)
+        if self.kernel in self.variable_length_kernels:
+            assert self.kernel == "gak"
+            estimator_kernel = "precomputed"
```

Comment:
Can't we do something similar as here in neighbors.py? Instead of overwriting self.metric and then resetting it before the return statement, a new variable `kernel_metric` could be made and that could be passed to cdist

Reply:
Well, the issue is that for kNN we do not have an external estimator that we fit, but instead we fit `self`.